### PR TITLE
Add service account example

### PR DESCRIPTION
**Pull request by dermesser**
(I'm still creating PRs so you know what's going on -- otherwise I can just push directly)

### CODE REVIEW

File: examples/service_account/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            }
+        }
+    }
+}
+
+// Publish some message every 2 seconds.
+fn publish_stuff(methods: &PubsubMethods, message: &str) {
+    check_or_create_topic(&methods);
+
+    let message = pubsub::PubsubMessage {
```
Comment (Byron) [inline examples/service_account/src/main.rs]: Could this also be the following ?
```Rust
let message = pubsub::PubsubMessage {
     // Base64 encoded!
    data: Some(base64::encode(message.as_bytes())),
    ...Default::default()
};
```

File: examples/service_account/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+fn ack_message(methods: &PubsubMethods, id: String) {
+    let request = pubsub::AcknowledgeRequest { ack_ids: Some(vec![id]) };
+    let result = methods.subscriptions_acknowledge(request, SUBSCRIPTION_NAME).doit();
+
+    match result {
+        Err(e) => {
+            // There's a JSON decode error happening, even on successful returns. :/
+            println!("Ack error: {:?}", e);
```
Comment (Byron) [inline examples/service_account/src/main.rs]: Hmm, maybe a problem in the code generated by google-apis-rs ? Or already fixed thanks to the `empty-structs` PR recently submitted ?

Comment (dermesser) [inline examples/service_account/src/main.rs]: Yes, the `empty-structs` PR has fixed this. Forgot to remove the comment though

File: examples/service_account/src/main.rs (hunk 1 above)
Comment (dermesser) [inline examples/service_account/src/main.rs]: good point!

**Comment by dermesser** [comment]
I addressed your points. I'll go ahead and use my newly acquired privilege, and merge this PR then :)

**Comment by Byron** [comment]
@dermesser Congratulations ! And thank you ! Just a moment ago I used the youtube command-line, not knowing what I was doing at all. The authentication was easy as pie, and the CLI program actually helped me to figure out what kind of information it needed. It's way nicer than I had it in mind, and quite ready for general consumption I believe. Also thanks to you !

**Pull request merged**